zipline/algorithm.py

```diff
@@ -632,42 +632,96 @@ def run(self, data=None, overwrite_sim_params=True):
     def _write_and_map_id_index_to_sids(self, identifiers, as_of_date):
         # Build new Assets for identifiers that can't be resolved as
         # sids/Assets
-        identifiers_to_build = set()
-        next_sid = max(self.asset_finder.sids or (0,)) + 1
+        def is_unknown(asset_or_sid):
+            sid = op.index(asset_or_sid)
+            return self.asset_finder.retrieve_asset(
```

Comment:
Got it.  I think even with bundles, there will be example cases where people have a dataframe - should we add a unit test for this?